crates/ruff_linter/src/rules/ruff/rules/tuple_map_int_version_parsing.rs

```diff
@@ -0,0 +1,161 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{Expr, ExprAttribute, ExprCall, ExprName};
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for calls of the form `tuple(map(int, __version__.split(".")))`.
+///
+/// ## Why is this bad?
+/// `__version__` does not always contain integral-like elements.
+///
+/// ```python
+/// import matplotlib  # 3.9.1.post-1
+///
+/// # ValueError: invalid literal for int() with base 10: 'post1'
+/// tuple(map(int, matplotlib.__version__.split(".")))
+/// ```
+///
+/// See also [PEP 440].
+///
+/// ## Example
+/// ```python
+/// tuple(map(int, matplotlib.__version__.split(".")))
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import packaging.version as version
+///
+/// version.parse(matplotlib.__version__)
+/// ```
+///
+/// [PEP 440]: https://peps.python.org/pep-0440/
```

Comment:
If users follow this link, there's a warning up at the top of the page telling them that the PEP is a historical document rather than living documentation that's meant to be kept up-to-date. Let's link to the up-to-date docs instead, which are at https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers